src/Razor/test/RazorLanguage.Test/TestFiles/IntegrationTests/ComponentRuntimeCodeGenerationTest/BindToElementFallback_WithFormat_WritesAttributes/TestComponent.codegen.cs

```diff
@@ -25,6 +25,7 @@ protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTr
 #nullable disable
             , "MM/dd"));
             builder.AddAttribute(3, "onchange", Microsoft.AspNetCore.Components.EventCallback.Factory.CreateBinder(this, __value => CurrentDate = __value, CurrentDate, "MM/dd"));
+            builder.SetUpdatesAttributeName("value");
```

Comment:
Sorry for all the questions, just trying to gleam some clarity here. So now that `SetUpdatesAttributeName` is its own thing (instead of the "value" being on `AddAttribute`) does the client listen to DOM updates for "value" specifically and "do the right thing"?

Reply:
This info isn’t used by the client. It’s used by the server when it receives an event. It needs to know whether that event is part of a 2-way binding, as the diffing logic needs to account for the fact that the client’s state (e.g. DOM element) was already mutated when the user edited its value.

Reply:
Sure but how do you know which attribute impacts which other attribute? I assume you don't need that level of granularity? 

Aka if you had two different attributes that were two way bound with different attributes:
```
<input @bind="..." @bind-somethingElse="..." ... />
```

Reply:
> Sure but how do you know which attribute impacts which other attribute

That's exactly what this information specifies. `SetUpdatesAttributeName` refers to the preceding event handler attribute (the "change" part of the two-way binding pair), and specifies which other attribute is the "value" part of the two-way binding pair.

> Aka if you had two different attributes that were two way bound with different attributes:

The compiler would emit two different `SetUpdatesAttributeName` calls, one after each of the corresponding event handler attributes.

Reply:
Ah, I see so the `SetUpdatesAttributeName` will impact the last attribute built. 

I'm not a huge fan of that super-stateful model. Would rather that be part of the `AddAttribute` call or have attributes have their own builder. I won't push hard on that though.

Reply:
I understand. I consider the statefulness as a least-worst option here, since doubling the existing large number of `AddAttribute` overloads gets us into a pretty bad place, and certainly will do if we keep adding features like this.

Since `RenderTreeBuilder` it's so low-level and its usage is primarily compiler generated, I think it's the best way forwards in this case. Even for use by humans we can make this safe, as we can strictly check the validity of these calls and throw if you try to call it at the wrong time.